src/bindings/js.ts

```diff
@@ -883,7 +883,10 @@ export class JSBuilder extends ExportsWalker {
       sb.push(`
 } = await (async url => instantiate(
   await (async () => {
-    try { return await globalThis.WebAssembly.compileStreaming(globalThis.fetch(url)); }
+    try { 
+      const source = await globalThis.fetch(url)
+      return await globalThis.WebAssembly.compileStreaming(source); 
+    }
```

Comment:
The main advantage of passing unresolved `fetch` in `WebAssembly.compileStreaming` is streaming compilation. With your approach, it's not better than use `WebAssembly.compile`. So I guess you should find another workaround for NextJS. Did you try use more more fresh version of node-fetch? NextJS uses `2.6.7` while least is `3.2.10`

Reply:
Got it, thanks! Overriding the node-fetch version used by Next seems to have done the trick. Thanks so much!